File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int left = 0;
        int right = k - 1;
        
        double largest = 0;
        if (nums.size() == 1)
            return nums[0];
        for(int i = 0; i < k; i++)
            largest += nums[i];
        
        double total = largest;
            
        while(right < nums.size()){
            right++;
            total += nums[right];
            
            total -= nums[left];
            left++;
            
            if(total > largest)
                largest = total; 
        }
        return largest/k;
    }
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        left = 0
        right = k-1
        
        largest = 0
        
        for i in range(k):
            print (nums[i])
            largest += nums[i]
        total = largest
        while right < len(nums)-1:   
            right += 1
            total += nums[right]
            
            total -= nums[left]
            left += 1
            if total > largest:
                largest = total
                
        return largest/k
            
            
        